README.md
# All Blue

A VS Code theme that's a little too obsessed with a certain primary color.

## Screenshots

Default theme
![All Blue screenshot](img/default-preview.png "All Blue screenshot")

Calm
![All Blue screenshot](img/calm-preview.png "Calm screenshot")

Waves
![All Blue screenshot](img/waves-preview.png "Waves screenshot")

The editor font used is [JetBrains Mono](https://www.jetbrains.com/lp/mono/)
with ligatures turned off.
The file icons are [Material Icons](https://marketplace.visualstudio.com/items?itemName=PKief.material-icon-theme).

I'm also shilling my other projects from the screenshots:

- [My resume site](https://github.com/njshockey/njshockey-resume)
- [Keysmith](https://github.com/njshockey/keysmith-rs), a Rust keygen library
- [BST Graph](https://github.com/njshockey/bst-graph),
    create Pokemon Base Stat Total charts.

## Syntax Colors

![Syntax colors](img/all-blue-colors.png)

The font used is [Cabin SemiCondensed](https://fonts.google.com/specimen/Cabin?query=cabin).
The Hex code font is [JetBrains Mono](https://www.jetbrains.com/lp/mono/).

For more details, head to [colors.md](colors.md).

## Installation

In the extensions panel in VS Code, type in `njshockey.all-blue` or visit one of the following links.

- [VS Marketplace (recommended)](https://marketplace.visualstudio.com/items?itemName=njshockey.all-blue)
- [Open VSX Registry](https://open-vsx.org/extension/njshockey/all-blue)

### Manual

To install this extension manually, download it from one of the following places:

- [GitHub releases](https://github.com/njshockey/all-blue-theme/releases)
- [VS Marketplace](https://marketplace.visualstudio.com/items?itemName=njshockey.all-blue)
  - This is under resources -> Download Extension
- [Open VSX Registry](https://open-vsx.org/extension/njshockey/all-blue)
  - This is under resources -> Download
- [Itch](https://njshockey.itch.io/all-blue)

Then, in the extensions panel, click on the three dots(...) left of your open tabs, then click on `Install from VSIX...`.

## Demo

For a downloadable demo, check out the [All Blue demo](https://github.com/njshockey/all-blue-demo).
Or just clone it with:

```bash
git clone https://github.com/njshockey/all-blue-demo.git
```

## Building

1. Clone this repo. Usually with `git clone https://github.com/njshockey/all-blue-theme.git`.
2. Run `yarn` to download the necessary NPM packages.
3. Open this repo in [VS Code](https://code.visualstudio.com/).
   - Make changes to the base colors in `./src/colors`.
   - Make changes to the templates in `./src/templates/editorTemplate.ts`
and `./src/templates/syntaxTemplate.ts`.
4. Run `yarn build-themes` to generate the JSON theme files
and press F5 to test changes.
    - Run `yarn build-themes-ugly` to build the themes without Prettier.
This should save you a few seconds.
5. To build the extension file (VSIX), install VSCE with
`npm install -g @vscode/vsce` then run `yarn build-ext`.

## Image credit

[Temporary icon](https://www.pexels.com/photo/white-sailboat-on-water-273886/)
by Pixabay.


CHANGELOG.md
# Changelog

For the source of each release, see [releases](https://github.com/njshockey/all-blue-theme/releases).

## v1.2.0 (Aug. 8, 2023)

Mostly a maintenance release, but with a few nice tweaks for better readability.

- Adjusted Status Bar. #002540 -> #001E33
- Adjusted sidebar section headers. #002540 -> #001E33
- Active tab background now uses what was previously the status bar background color for better contrast. #002138 -> #002540
- Adjusted inlay hints
  - Foreground color. #ffffff -> #006680
  - Background color. #001E33 -> #002138
  - Parameter foreground. #0892D0 -> #006680
- Adjusted editor suggest widget selected foreground. #003459 -> #003A63
- Definition keywords now use the operator color (#00CCFF, #8BC6FC) to be more consistent with JSON.
  - This effects several file types like editorconfig, toml, gitattributes, etc.
- Several Rust keywords now use the operator color.
- Added download links to the readme.

## v1.1.1 (Feb. 9, 2023)

- Active line number now uses the foreground color. #D8DEE9 -> #63C5DA
- Suggest widget is now themed.
- Codelens foreground is now themed.
- Settings side bar headings have a better highlight color.
- Settings focused headings are much more visible.

## v1.1.0 (Feb. 4, 2023)

- The themes are now generated through TypeScript instead of
being manually edited by hand.
    - This brings a lot more consistency and fixes a lot of minor issues.
    - Also it's way less annoying to make changes. (The important part.)
- Terminal colors are now themed.
- Swapped around editor background colors.
    - Editor background:  #002540 -> #002138
    - Side Bar background: #002138 -> #001E33
    - Status Bar background: #001E33 -> #002540
    - Current line: #003154 -> #002C4C
- Changed Waves pink: #FF80DD -> #F77FBE
- Changed Calm yellow: #FFFC79 -> #FEF08A
- Changed error/deleted color from Calm red to Waves red.
- Booleans and numbers are now *italic*.
- Tweaked All Blue bracket colors.
- Properties now use the tag color. (Red in variants.)
- Escape characters now use the function color. (Blue in variants.)
- JS Constants and Objects now use the component color. (Green in variants.)
- Tabs now have a bottom border when active.
- Reworked selection highlighting.
- Keybindings menu is now themed.
- Along with other various changes to make the look and feel more cohesive.

## v1.0.2 (Jan. 25, 2023)

- Improved editor word highlighting.
- Reduced extension size by removing screenshots from VSIX.

## v1.0.1 (Jan. 23, 2023)

- Debug toolbar is now themed in Waves and Calm.

## v1.0.0 (Jan. 22, 2023)

- Initial release


src/main.ts
import { writeFile, existsSync, mkdirSync } from "fs";
import { createFullTemplate } from "./templates/fullTemplate";
import { Themes } from "./colors";

function createJson() {
	const themes: Themes[] = [Themes.AllBlue, Themes.Calm, Themes.Waves];
	const outputDir = "./themes";

	if (!existsSync(outputDir)) {
		mkdirSync(outputDir, { recursive: true });
	}

	for (let i = 0; i < themes.length; i++) {
		const output = createFullTemplate(themes[i]);
		let fileName = "";

		switch (themes[i]) {
			case Themes.AllBlue:
				fileName = "all-blue";
				break;
			case Themes.Calm:
				fileName = "all-blue-calm";
				break;
			case Themes.Waves:
				fileName = "all-blue-waves";
				break;
		}

		fileName += "-color-theme.json";

		writeFile(outputDir + "/" + fileName, output, (err) => {
			if (!err) {
				console.log(`Successfully wrote ${fileName}.`);
			} else {
				console.error(err);
			}
		});
	}
}

createJson();


src/templates/editorTemplate.ts
// Template for creating the editor portion of the theme

import {
	Themes,
	editorColors as eC,
	terminalColors as terminal,
} from "../colors";
import { checkTheme } from "./checkTheme";

export function createEditorTemplate(theme: Themes) {
	const syntax = checkTheme(theme);

	const output = {
		name: "All Blue",
		colors: {
			// Editor
			"errorForeground": eC.error,
			"editor.background": eC.bgEditor,
			"editor.foreground": eC.fg,
			"editor.lineHighlightBackground": eC.bgCurrentLine,
			"editor.foldBackground": eC.bgEditor,
			"editorCursor.foreground": eC.cursor,
			"editor.selectionBackground": eC.selectionBright,
			"editor.wordHighlightBackground": eC.selectionHighlight,
			"editor.wordHighlightStrongBackground": eC.selectionHighlight,
			"editor.wordHighlightStrongBorder":
				eC.selectionHighlightStrongBorder,
			"editor.findMatchBackground": eC.selectionHighlight,
			"editor.findMatchBorder": eC.selectionHighlightBorder,
			"editor.findMatchHighlightBackground": eC.selectionHighlight,
			"editor.hoverHighlightBackground": eC.selectionHighlight,
			"editorLink.activeForeground": eC.linkEditor,
			"editor.rangeHighlightBackground": eC.selectionHighlight,
			"editorGroup.dropBackground":
				eC.selectionBright + eC.selectionTransparency,
			"editorGroupHeader.tabsBackground": eC.bgSideBar,
			"editorGroup.border": eC.borderDim,
			"editorInfo.foreground": eC.info,
			"editorWarning.foreground": eC.warning,
			"editorError.foreground": eC.error,

			// Gutter
			"editorLineNumber.activeForeground": eC.lineNumberActiveFg,
			"editorLineNumber.foreground": eC.fgDisabled,
			"editorLightBulb.foreground": eC.lightbulb,
			"editorGutter.modifiedBackground": eC.modifiedGutter,
			"editorGutter.addedBackground": eC.addedGutter,
			"editorGutter.deletedBackground": eC.deletedGutter,
			"editorGutter.commentRangeForeground": eC.fgDisabled,

			// Indents and whitespace
			"editorIndentGuide.background1": eC.selectionNormal,
			"editorIndentGuide.activeBackground1": eC.fg,
			"editorWhitespace.foreground": eC.fgDisabled,

			// Hover widget
			"editorHoverWidget.background": eC.bgSideBar,
			"editorHoverWidget.border": eC.borderDim,
			"editorHoverWidget.foreground": eC.fgFourth,
			"editorHoverWidget.statusBarBackground": eC.bgStatusBar,

			// Editor widget
			"editorWidget.background": eC.bgSideBar,
			"editorWidget.foreground": eC.fg,
			"editorWidget.resizeBorder": eC.borderBright,
			"editorSuggestWidget.foreground": eC.fg,
			"editorSuggestWidget.background": eC.bgStatusBar,
			"editorSuggestWidget.border": eC.borderDim,
			"editorSuggestWidget.selectedBackground": eC.selectionBright,
			"editorSuggestWidget.selectedForeground": eC.fgTertiary,
			"editorSuggestWidget.highlightForeground": eC.fgFourth,
			"editorSuggestWidget.focusHighlightForeground": eC.fgFourth,
			"editorGhostText.foreground": eC.fgDisabled,

			// Inlay hints
			"editorInlayHint.background": eC.bgEditor,
			"editorInlayHint.foreground": eC.fgDisabled,
			"editorInlayHint.typeForeground": eC.fgFourth,
			"editorInlayHint.parameterForeground": eC.inlayHintParameterFg,
			"editorInlayHint.parameterBackground": eC.bgEditor,
			"editorCodeLens.foreground": eC.fgDisabled,

			// Breadcrumbs
			"breadcrumb.foreground": eC.fg,
			"breadcrumb.focusForeground": "#ffffff",
			"breadcrumb.activeSelectionForeground": "#ffffff",
			"breadcrumbPicker.background": eC.bgSideBar,

			// Peek view
			"peekView.border": eC.borderDim,
			"peekViewEditor.background": eC.bgSideBar,
			"peekViewEditor.matchHighlightBackground": eC.selectionBright,
			"peekViewResult.background": eC.bgSideBar,
			"peekViewResult.fileForeground": eC.fg,
			"peekViewTitle.background": eC.bgEditor,
			"peekViewTitleLabel.foreground": eC.fg,
			"peekViewTitleDescription.foreground": eC.peekPreviewTitle,

			// Side bar
			"sideBarTitle.foreground": eC.fg,
			"sideBar.background": eC.bgSideBar,
			"sideBar.foreground": eC.fg,
			"sideBar.dropBackground":
				eC.selectionBright + eC.selectionTransparency,
			"sideBarSectionHeader.background": eC.bgSideBar,
			"sash.hoverBorder": eC.borderBright,
			"focusBorder": eC.borderDim,

			// Activity bar
			"activityBarBadge.background": eC.badges,
			"activityBarBadge.foreground": "#ffffff",
			"activityBar.background": eC.bgSideBar,
			"activityBar.foreground": eC.fgFourth,

			// Status bar
			"statusBar.background": eC.bgSideBar,
			"statusBar.foreground": eC.fg,
			"statusBar.focusBorder": eC.transparent,
			"statusBar.noFolderBackground": eC.bgSideBar,
			"statusBar.noFolderForeground": eC.fg,
			"statusBar.debuggingBackground": "#004B80",
			"statusBar.debuggingForeground": eC.fgSecondary,
			"statusBarItem.remoteBackground": eC.fgDisabled,
			"statusBarItem.remoteForeground": eC.fgSecondary,
			"statusBarItem.hoverBackground": eC.selectionNormal,
			"statusBarItem.activeBackground": eC.selectionBright,
			"statusBarItem.focusBorder": eC.transparent,

			// Menu
			"menubar.selectionBackground": eC.selectionNormal,
			"menubar.selectionForeground": eC.fgSecondary,
			"menu.background": eC.bgEditor,
			"menu.foreground": eC.fg,
			"menu.selectionBackground": eC.selectionNormal,
			"menu.selectionForeground": eC.fgSecondary,
			"menu.separatorBackground": eC.borderDim,

			// Title bar
			"titleBar.activeBackground": eC.bgSideBar,
			"titleBar.inactiveBackground": eC.bgEditor,
			"titleBar.activeForeground": eC.fg,
			"titleBar.inactiveForeground": eC.fgDisabled,

			// Tabs
			"tab.inactiveBackground": eC.bgSideBar,
			"tab.inactiveForeground": eC.fgDisabled,
			"tab.activeForeground": eC.fgSecondary,
			"tab.activeBackground": eC.bgStatusBar,
			"tab.border": eC.transparent,
			"tab.activeBorder": eC.fgFourth,
			"tab.activeModifiedBorder": eC.borderBright,
			"tab.inactiveModifiedBorder": eC.borderBright,

			// Scrollbar
			"scrollbarSlider.background":
				eC.selectionBright + eC.selectionTransparency,
			"scrollbarSlider.hoverBackground":
				"#004b80" + eC.selectionTransparency,
			"scrollbarSlider.activeBackground":
				eC.modifiedGutter + eC.selectionTransparency,

			// Editor overview ruler
			"editorOverviewRuler.addedForeground": eC.addedGutter,
			"editorOverviewRuler.modifiedForeground": eC.modifiedGutter,
			"editorOverviewRuler.deletedForeground": eC.deletedGutter,
			"editorOverviewRuler.errorForeground": eC.error,
			"editorOverviewRuler.infoForeground": eC.info,
			"editorOverviewRuler.warningForeground": eC.warning,
			"editorOverviewRuler.bracketMatchForeground": eC.selectionHighlight,
			"editorOverviewRuler.findMatchForeground": eC.selectionHighlight,
			"editorOverviewRuler.rangeHighlightForeground":
				eC.selectionHighlight,
			"editorOverviewRuler.selectionHighlightForeground":
				eC.selectionHighlight,
			"editorOverviewRuler.wordHighlightForeground":
				eC.selectionHighlight,
			"editorOverviewRuler.wordHighlightStrongForeground":
				eC.selectionHighlightStrongBorder,
			"editorOverviewRuler.border": eC.transparent,

			// Minimap
			"minimap.warningHighlight": eC.warning,
			"minimap.errorHighlight": eC.error,
			"minimap.selectionHighlight": eC.selectionHighlight,
			"minimap.findMatchHighlight": eC.selectionHighlight,
			"minimap.selectionOccurrenceHighlight": eC.selectionHighlight,
			"minimapGutter.addedBackground": eC.addedGutter,
			"minimapGutter.deletedBackground": eC.deletedGutter,
			"minimapGutter.modifiedBackground": eC.modifiedGutter,

			// Lists
			"list.hoverBackground": eC.bgStatusBar,
			"list.dropBackground":
				eC.selectionBright + eC.selectionTransparency,
			"list.inactiveSelectionBackground": eC.bgEditor,
			"list.activeSelectionForeground": eC.fgSecondary,
			"list.inactiveFocusOutline": eC.bgEditor,
			"list.focusOutline": eC.bgEditor,
			"list.warningForeground": eC.warning,
			"list.errorForeground": eC.error,
			"tree.indentGuidesStroke": "#004B80",

			// Git decorations
			"gitDecoration.untrackedResourceForeground": eC.added,
			"gitDecoration.modifiedResourceForeground": eC.modified,
			"gitDecoration.renamedResourceForeground": eC.modified,
			"gitDecoration.deletedResourceForeground": eC.deleted,
			"gitDecoration.conflictingResourceForeground": eC.deleted,
			"gitDecoration.stageDeletedResourceForeground": eC.deleted,
			"gitDecoration.ignoredResourceForeground": eC.ignored,
			"gitDecoration.addedResourceForeground": eC.staged,
			"gitDecoration.stageModifiedResourceForeground": eC.staged,

			// Settings
			"settings.textInputBackground": eC.bgSideBar,
			"settings.textInputForeground": eC.fg,
			"settings.textInputBorder": eC.borderDim,
			"settings.numberInputBackground": eC.bgSideBar,
			"settings.numberInputForeground": eC.fg,
			"settings.numberInputBorder": eC.borderDim,
			"settings.checkboxBackground": eC.bgSideBar,
			"settings.checkboxForeground": eC.fg,
			"settings.checkboxBorder": eC.borderDim,
			"settings.dropdownBackground": eC.bgSideBar,
			"settings.dropdownForeground": eC.fg,
			"settings.dropdownBorder": eC.borderDim,
			"settings.dropdownListBorder": eC.borderDim,
			"settings.sashBorder": eC.borderDim,
			"settings.headerBorder": eC.borderDim,
			"settings.headerForeground": eC.fgSecondary,
			"settings.modifiedItemIndicator": eC.cursor,
			"settings.rowHoverBackground": eC.bgSideBar,
			"settings.focusedRowBorder": eC.borderDim,
			"settings.focusedRowBackground": eC.bgSideBar,
			"settings.settingsHeaderHoverForeground": eC.fgSecondary,

			// Text
			"foreground": eC.fg,
			"descriptionForeground": eC.fg,
			"icon.foreground": eC.fg,
			"textBlockQuote.background": eC.bgSideBar,
			"textBlockQuote.border": eC.borderBright,
			"textCodeBlock.background": eC.bgSideBar,
			"textLink.foreground": eC.cursor,
			"textLink.activeForeground": eC.linkText,
			"textPreformat.foreground": eC.textPreformat,
			"textSeparator.foreground": eC.fgDisabled,

			// Extensions
			"extensionIcon.verifiedForeground": eC.cursor,
			"extensionButton.background": eC.selectionBright,
			"extensionButton.foreground": eC.fgSecondary,
			"extensionButton.separator": eC.fgDisabled,

			// Welcome page and walkthrough
			"welcomePage.tileBackground": eC.bgSideBar,
			"welcomePage.tileHoverBackground": eC.selectionNormal,
			"welcomePage.progress.foreground": eC.cursor,
			"walkthrough.stepTitle.foreground": eC.fg,
			"walkThrough.embeddedEditorBackground": eC.bgSideBar,

			// Command Center
			"commandCenter.foreground": eC.fg,
			"commandCenter.background": eC.bgSideBar,
			"commandCenter.inactiveBorder": eC.borderInactive,
			"commandCenter.activeBackground": eC.selectionNormal,
			"commandCenter.activeBorder": eC.borderDim,
			"commandCenter.activeForeground": eC.linkText,

			// Badges
			"badge.background": eC.badges,
			"badge.foreground": "#ffffff",

			// Input
			"input.background": eC.bgSideBar,
			"input.border": eC.borderInactive,
			"input.foreground": eC.fg,
			"input.placeholderForeground": eC.fgDisabled,
			"inputValidation.errorBackground": eC.bgEditor,
			"inputValidation.errorForeground": eC.fg,
			"inputValidation.errorBorder": eC.error,
			"inputValidation.infoBackground": eC.bgEditor,
			"inputValidation.infoForeground": eC.fg,
			"inputValidation.infoBorder": eC.info,
			"inputValidation.warningBackground": eC.bgEditor,
			"inputValidation.warningForeground": eC.fg,
			"inputValidation.warningBorder": eC.warning,
			"button.background": eC.selectionBright,
			"button.foreground": eC.fgSecondary,
			"button.separator": eC.fgDisabled,
			"dropdown.background": eC.bgSideBar,
			"dropdown.foreground": eC.fg,
			"selection.background": eC.selectionBright,

			// Quick input
			"pickerGroup.border": eC.borderDim,
			"pickerGroup.foreground": syntax.operator,
			"quickInput.background": eC.bgSideBar,
			"quickInput.foreground": eC.fg,
			"quickInputList.focusBackground": eC.selectionNormal,
			"quickInputList.focusForeground": eC.fgSecondary,
			"quickInputList.focusIconForeground": eC.fgSecondary,
			"quickInputTitle.background": eC.selectionNormal,

			// Panel
			"panel.border": eC.borderDim,
			"panelTitle.activeForeground": eC.fgSecondary,
			"panelTitle.inactiveForeground": eC.fg,
			"panelSection.dropBackground":
				eC.selectionBright + eC.selectionTransparency,
			"problemsInfoIcon.foreground": eC.info,
			"problemsWarningIcon.foreground": eC.warning,
			"problemsErrorIcon.foreground": eC.error,

			// Terminal
			"terminal.foreground": eC.fg,
			"terminalCursor.foreground": eC.cursor,
			"terminal.dropBackground":
				eC.selectionBright + eC.selectionTransparency,
			"terminal.selectionBackground": eC.selectionBright,
			"terminal.findMatchBackground": eC.selectionHighlight,
			"terminal.findMatchBorder": eC.selectionHighlightBorder,
			"terminal.inactiveSelectionBackground": eC.selectionDim,
			"terminalOverviewRuler.cursorForeground": eC.cursor,
			"terminalOverviewRuler.findMatchForeground": eC.selectionHighlight,
			"terminalCommandDecoration.defaultBackground": eC.fg,
			"terminalCommandDecoration.successBackground": eC.info,
			"terminalCommandDecoration.errorBackground": eC.error,

			// Terminal colors
			"terminal.ansiBlack": terminal.black,
			"terminal.ansiBrightBlack": terminal.brightBlack,
			"terminal.ansiWhite": terminal.white,
			"terminal.ansiBrightWhite": terminal.brightWhite,
			"terminal.ansiRed": terminal.red,
			"terminal.ansiBrightRed": terminal.brightRed,
			"terminal.ansiGreen": terminal.green,
			"terminal.ansiBrightGreen": terminal.brightGreen,
			"terminal.ansiBlue": terminal.blue,
			"terminal.ansiBrightBlue": terminal.brightBlue,
			"terminal.ansiYellow": terminal.yellow,
			"terminal.ansiBrightYellow": terminal.brightYellow,
			"terminal.ansiCyan": terminal.cyan,
			"terminal.ansiBrightCyan": terminal.brightCyan,
			"terminal.ansiMagenta": terminal.magenta,
			"terminal.ansiBrightMagenta": terminal.brightMagenta,

			// Keybindings
			"keybindingLabel.foreground": eC.fg,
			"keybindingLabel.background": eC.bgCurrentLine,
			"keybindingLabel.border": eC.borderDim,
			"keybindingTable.rowsBackground": eC.bgSideBar,
			"keybindingTable.headerBackground": eC.bgCurrentLine,

			// Debug tool bar
			"debugToolBar.background": eC.selectionDim,
			"debugIcon.stopForeground": eC.error,
			"debugIcon.disconnectForeground": eC.error,
			"debugIcon.restartForeground": eC.added,
			"debugIcon.pauseForeground": eC.fgFourth,
			"debugIcon.continueForeground": eC.fgFourth,
			"debugIcon.stepBackForeground": eC.fgFourth,
			"debugIcon.stepIntoForeground": eC.fgFourth,
			"debugIcon.stepOutForeground": eC.fgFourth,
			"debugIcon.stepOverForeground": eC.fgFourth,
			"debugIcon.startForeground": eC.fgFourth,
			"debugIcon.breakpointForeground": eC.fgFourth,
			"debugIcon.breakpointCurrentStackframeForeground": eC.fgFourth,
			"debugIcon.breakpointStackframeForeground": eC.fgFourth,
			"debugIcon.breakpointUnverifiedForeground": eC.fgFourth,
			"debugIcon.breakpointDisabledForeground": eC.fgDisabled,

			// Bracket highlight
			"editorBracketHighlight.foreground1": syntax.bracket1,
			"editorBracketHighlight.foreground2": syntax.bracket2,
			"editorBracketHighlight.foreground3": syntax.bracket3,
			"editorBracketHighlight.foreground4": syntax.bracket4,
			"editorBracketHighlight.foreground5": syntax.bracket5,
			"editorBracketHighlight.foreground6": syntax.bracket6,
			"editorBracketHighlight.unexpectedBracket.foreground": eC.error,
			"editorBracketMatch.background": eC.selectionHighlight,
			"editorBracketMatch.border": eC.transparent,
		},
	};

	return output;
}


src/colors.ts
export enum Themes {
	AllBlue,
	Waves,
	Calm,
}

// Declare colors for syntax highlighting
export interface SyntaxColors {
	// Comments
	comment: string;

	// Operator, modifier, units
	operator: string;

	// Variables, CSS ID's
	variable: string;

	// Numbers, booleans, null, this
	bool: string;

	// Types, attributes, classes
	type: string;

	// Functions, Pseudo-classes
	function: string;

	// HTML tags, escape chars
	tag: string;

	// Strings
	string: string;

	// JS Components, Regex
	component: string;

	// Bracket highlighting
	bracket1: string;
	bracket2: string;
	bracket3: string;
	bracket4: string;
	bracket5: string;
	bracket6: string;
}

// Generic base colors
interface SyntaxColorsGeneric {
	red: string;
	blue: string;
	cyan: string;
	green: string;
	yellow: string;
	orange: string;
	pink: string;
	grey: string;
	beige: string;
}

export const allBlueGeneric = {
	blue: "#4F86F7",
	darkBlue: "#1974D2",
	lightBlue: "#a5b4fc",
	teal: "#06B6D4",
	darkTeal: "#0892D0",
	lightTeal: "#AEC6CF",
	blueGrey: "#006680",
	cyan: "#00CCFF",
	lilac: "#85A1F2",
};

export const allBlueColors: SyntaxColors = {
	comment: allBlueGeneric.blueGrey,
	operator: allBlueGeneric.cyan,
	variable: allBlueGeneric.lilac,
	bool: allBlueGeneric.darkBlue,
	type: allBlueGeneric.blue,
	function: allBlueGeneric.lightBlue,
	tag: allBlueGeneric.darkTeal,
	string: allBlueGeneric.lightTeal,
	component: allBlueGeneric.teal,
	bracket1: allBlueGeneric.cyan,
	bracket2: allBlueGeneric.blue,
	bracket3: allBlueGeneric.lilac,
	bracket4: "#63C5DA",
	bracket5: allBlueGeneric.lightBlue,
	bracket6: allBlueGeneric.darkBlue,
};

export const wavesGeneric: SyntaxColorsGeneric = {
	red: "#FF7566",
	blue: "#079BF5",
	green: "#0BDA51",
	yellow: "#FFFD37",
	orange: "#FFAD00",
	pink: "#F77FBE",
	cyan: allBlueGeneric.cyan,
	grey: allBlueGeneric.blueGrey,
	beige: "#DBD59D",
};

export const wavesColors: SyntaxColors = {
	comment: wavesGeneric.grey,
	operator: wavesGeneric.cyan,
	variable: wavesGeneric.orange,
	bool: wavesGeneric.pink,
	type: wavesGeneric.yellow,
	function: wavesGeneric.blue,
	tag: wavesGeneric.red,
	string: wavesGeneric.beige,
	component: wavesGeneric.green,
	bracket1: wavesGeneric.cyan,
	bracket2: wavesGeneric.yellow,
	bracket3: "#63C5DA",
	bracket4: wavesGeneric.orange,
	bracket5: wavesGeneric.blue,
	bracket6: wavesGeneric.green,
};

export const calmGeneric: SyntaxColorsGeneric = {
	red: "#F9968B",
	blue: "#85A1F2",
	green: "#50C878",
	yellow: "#FEF08A",
	orange: "#FFAC40",
	pink: "#FC8EAC",
	cyan: "#8BC6FC",
	grey: allBlueGeneric.blueGrey,
	beige: "#D5C2A5",
};

export const calmColors: SyntaxColors = {
	comment: calmGeneric.grey,
	operator: calmGeneric.cyan,
	variable: calmGeneric.orange,
	bool: calmGeneric.pink,
	type: calmGeneric.yellow,
	function: calmGeneric.blue,
	tag: calmGeneric.red,
	string: calmGeneric.beige,
	component: calmGeneric.green,
	bracket1: calmGeneric.cyan,
	bracket2: calmGeneric.yellow,
	bracket3: "#63C5DA",
	bracket4: calmGeneric.orange,
	bracket5: calmGeneric.blue,
	bracket6: calmGeneric.green,
};

export const terminalColors = {
	black: "#000000",
	brightBlack: "#2E3440",
	white: "#E5E9F0",
	brightWhite: "#FFFFFF",
	red: calmGeneric.red,
	brightRed: wavesGeneric.red,
	green: calmGeneric.green,
	brightGreen: wavesGeneric.green,
	blue: allBlueColors.type,
	brightBlue: wavesGeneric.blue,
	yellow: calmGeneric.yellow,
	brightYellow: wavesGeneric.yellow,
	cyan: allBlueColors.string,
	brightCyan: wavesGeneric.cyan,
	magenta: calmGeneric.pink,
	brightMagenta: wavesGeneric.pink,
};

const editorColorsGeneric = {
	fg: "#63C5DA",
	fgSecondary: "#eceff4",
	fgTertiary: "#D8DEE9",
	fgFourth: allBlueColors.operator,
	fgDisabled: calmColors.comment,
	bgEditor: "#002138",
	bgSideBar: "#001E33",
	bgStatusBar: "#002540",
	bgCurrentLine: "#002C4C",
	transparent: "#00000000",
	cursor: "#0094FF",
	lightbulb: calmGeneric.yellow,
	badges: "#007FFF",
	info: calmGeneric.blue,
	warning: calmGeneric.orange,
	error: wavesGeneric.red,
	added: calmGeneric.green,
	modified: calmGeneric.blue,
	deleted: wavesGeneric.red,
	staged: calmGeneric.yellow,
	ignored: calmGeneric.grey,
	addedGutter: "#2E8B57",
	modifiedGutter: "#005999",
	deletedGutter: wavesGeneric.red,
	selectionNormal: "#003459",
	selectionDim: "#002E4F",
	selectionBright: "#003A63",
	selectionVeryBright: "#00528C",
	selectionTransparency: "8e", // Hex transparency. Ex: "8e"
	selectionHighlightBorder: allBlueColors.operator,
	selectionHighlightStrongBorder: allBlueColors.type,
	linkEditor: allBlueColors.tag,
	linkText: allBlueColors.string,
	textPreformat: calmGeneric.blue,
};

export const editorColors = {
	fg: editorColorsGeneric.fg,
	fgSecondary: editorColorsGeneric.fgSecondary,
	fgTertiary: editorColorsGeneric.fgTertiary,
	fgFourth: editorColorsGeneric.fgFourth,
	fgDisabled: editorColorsGeneric.fgDisabled,
	bgEditor: editorColorsGeneric.bgEditor,
	bgSideBar: editorColorsGeneric.bgSideBar,
	bgStatusBar: editorColorsGeneric.bgStatusBar,
	bgCurrentLine: editorColorsGeneric.bgCurrentLine,
	transparent: editorColorsGeneric.transparent,
	cursor: editorColorsGeneric.cursor,
	lightbulb: editorColorsGeneric.lightbulb,
	badges: editorColorsGeneric.badges,
	borderInactive: editorColorsGeneric.bgCurrentLine,
	borderDim: editorColorsGeneric.selectionNormal,
	borderBright: editorColorsGeneric.cursor,
	selectionNormal: editorColorsGeneric.selectionNormal,
	selectionDim: editorColorsGeneric.selectionDim,
	selectionBright: editorColorsGeneric.selectionBright,
	selectionVeryBright: editorColorsGeneric.selectionVeryBright,
	selectionTransparency: editorColorsGeneric.selectionTransparency,
	// Selection for word highlighting and find match
	selectionHighlight:
		editorColorsGeneric.selectionVeryBright +
		editorColorsGeneric.selectionTransparency,
	selectionHighlightBorder: editorColorsGeneric.selectionHighlightBorder,
	selectionHighlightStrongBorder:
		editorColorsGeneric.selectionHighlightStrongBorder,
	info: editorColorsGeneric.info,
	warning: editorColorsGeneric.warning,
	error: editorColorsGeneric.error,
	added: editorColorsGeneric.added,
	modified: editorColorsGeneric.modified,
	deleted: editorColorsGeneric.deleted,
	staged: editorColorsGeneric.staged,
	ignored: editorColorsGeneric.ignored,
	addedGutter: editorColorsGeneric.addedGutter,
	modifiedGutter: editorColorsGeneric.modifiedGutter,
	deletedGutter: editorColorsGeneric.deletedGutter,
	lineNumberActiveFg: editorColorsGeneric.fg,
	inlayHintParameterFg: allBlueColors.comment,
	peekPreviewTitle: allBlueColors.comment,
	linkEditor: editorColorsGeneric.linkEditor,
	linkText: editorColorsGeneric.linkText,
	textPreformat: editorColorsGeneric.textPreformat,
};
